src/apply.jl

```diff
@@ -125,3 +126,24 @@ function apply_append(table, t; kwargs...)
     result = Tables.columntable(apply(table, t; kwargs...))
     return T(merge(Tables.columntable(table), result))
 end
+
+
+# These methods format data according to the cardinality of the Transform.
+# Most Transforms don't require any formatting, only those that are ManyToOne do.
+# Note: we don't yet have a ManyToMany transform, so those might need separate treatment.
+
+# _preformat formats the data before calling _apply. Needed for all apply methods.
+# Before applying a ManyToOne Transform we must first slice up the data along the dimension
+# we are reducing over.
+_preformat(::Cardinality, A, d) = A
+_preformat(::ManyToOne, A, d) = eachslice(A; dims=d)
+
+# _postformat formats the data after calling _apply. Needed for apply_append.
+# After applying a ManyToOne Transform, if we want to cat the result we have to reshape it
+# setting the reduced dimension to 1, otherwise cat will error.
+_postformat(::Cardinality, result, A, d) = result
+function _postformat(::ManyToOne, result, A, d)
```

Comment:
sure, I renamed `d`->`append_dim` in the other PR but I'll explain here as well.

Basically, the problem was when we want to append a row to an array, but LinearCombination always returns a column vector. So we have to reshape it into a row to get it to fit.

In general, when we want to `cat` the result with `apply_append` for ManyToOne transforms (which always return a N-1 array) we have to reshape it so that the reduced dimension is always 1 so that it fits.